File: java/semantickernel-core/src/test/java/com/microsoft/semantickernel/memory/MemoryExceptionTests.java
// Copyright (c) Microsoft. All rights reserved.
package com.microsoft.semantickernel.memory;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNull;
import static org.junit.jupiter.api.Assertions.assertSame;
import static org.junit.jupiter.api.Assertions.assertTrue;

import org.junit.jupiter.api.Test;

class MemoryExceptionTests {
    @Test
    void itRoundtripsArgsToErrorCodeCtor() {
        // Arrange
        MemoryException e =
                new MemoryException(MemoryException.ErrorCodes.FAILED_TO_CREATE_COLLECTION);

        // Assert
        assertEquals(MemoryException.ErrorCodes.FAILED_TO_CREATE_COLLECTION, e.getErrorCode());
        assertTrue(e.getMessage().contains("Failed to create collection"));
        assertNull(e.getCause());
    }

    @Test
    void itRoundtripsArgsToErrorCodeMessageCtor() {
        // Arrange
        String message = "this is a test";
        MemoryException e =
                new MemoryException(
                        MemoryException.ErrorCodes.FAILED_TO_CREATE_COLLECTION, message);

        // Assert
        assertEquals(MemoryException.ErrorCodes.FAILED_TO_CREATE_COLLECTION, e.getErrorCode());
        assertTrue(e.getMessage().contains("Failed to create collection"));
        assertTrue(e.getMessage().contains(message));
        assertNull(e.getCause());
    }

    @Test
    void itRoundtripsArgsToErrorCodeMessageExceptionCtor() {
        // Arrange
        String message = "this is a test";
        Exception inner = new NumberFormatException();
        MemoryException e =
                new MemoryException(
                        MemoryException.ErrorCodes.FAILED_TO_CREATE_COLLECTION, message, inner);

        // Assert
        assertEquals(MemoryException.ErrorCodes.FAILED_TO_CREATE_COLLECTION, e.getErrorCode());
        assertTrue(e.getMessage().contains("Failed to create collection"));
        assertTrue(e.getMessage().contains(message));
        assertSame(inner, e.getCause());
    }

    @Test
    void itAllowsNullMessageAndInnerExceptionInCtors() {
        // Arrange
        MemoryException e =
                new MemoryException(
                        MemoryException.ErrorCodes.FAILED_TO_CREATE_COLLECTION, null, null);

        // Assert
        assertEquals(MemoryException.ErrorCodes.FAILED_TO_CREATE_COLLECTION, e.getErrorCode());
        assertTrue(e.getMessage().contains("Failed to create collection"));
        assertNull(e.getCause());
    }
}

File: dotnet/src/Plugins/Memory/Plugins.Memory.UnitTests/MemoryExceptionTests.cs
// Copyright (c) Microsoft. All rights reserved.

using System;
using Microsoft.SemanticKernel.Plugins.Memory;
using Xunit;

namespace SemanticKernel.Plugins.Memory.UnitTests;

public class MemoryExceptionTests
{
    [Fact]
    public void ItRoundtripsArgsToErrorCodeCtor()
    {
        // Arrange
        var e = new MemoryException(MemoryException.ErrorCodes.FailedToCreateCollection);

        // Assert
        Assert.Equal(MemoryException.ErrorCodes.FailedToCreateCollection, e.ErrorCode);
        Assert.Contains("Failed to create collection", e.Message, StringComparison.Ordinal);
        Assert.Null(e.InnerException);
    }

    [Fact]
    public void ItRoundtripsArgsToErrorCodeMessageCtor()
    {
        // Arrange
        const string Message = "this is a test";
        var e = new MemoryException(MemoryException.ErrorCodes.FailedToCreateCollection, Message);

        // Assert
        Assert.Equal(MemoryException.ErrorCodes.FailedToCreateCollection, e.ErrorCode);
        Assert.Contains("Failed to create collection", e.Message, StringComparison.Ordinal);
        Assert.Contains(Message, e.Message, StringComparison.Ordinal);
        Assert.Null(e.InnerException);
    }

    [Fact]
    public void ItRoundtripsArgsToErrorCodeMessageExceptionCtor()
    {
        // Arrange
        const string Message = "this is a test";
        var inner = new FormatException();
        var e = new MemoryException(MemoryException.ErrorCodes.FailedToCreateCollection, Message, inner);

        // Assert
        Assert.Equal(MemoryException.ErrorCodes.FailedToCreateCollection, e.ErrorCode);
        Assert.Contains("Failed to create collection", e.Message, StringComparison.Ordinal);
        Assert.Contains(Message, e.Message, StringComparison.Ordinal);
        Assert.Same(inner, e.InnerException);
    }

    [Fact]
    public void ItAllowsNullMessageAndInnerExceptionInCtors()
    {
        // Arrange
        var e = new MemoryException(MemoryException.ErrorCodes.FailedToCreateCollection, null, null);

        // Assert
        Assert.Equal(MemoryException.ErrorCodes.FailedToCreateCollection, e.ErrorCode);
        Assert.Contains("Failed to create collection", e.Message, StringComparison.Ordinal);
        Assert.Null(e.InnerException);
    }
}
